account-comparison/programs/account-comparison/src/lib.rs

```diff
@@ -61,47 +81,57 @@ pub mod account_comparison {
             &crate::ID,
         );
 
-        // LightAccount::new_init will create an account with empty output state (no input state).
-        // Modifying the account will modify the output state that when converted to_account_info()
-        // is hashed with poseidon hashes, serialized with borsh
-        // and created with invoke_light_system_program by invoking the light-system-program.
-        // The hashing scheme is the account structure derived with LightHasher.
+        // Create a new compressed account with empty input state
+        // LightAccount::new_init initializes an account for creation
+        // The account data will be hashed using Poseidon hashes and stored off-chain
         let mut compressed_account = LightAccount::<'_, CompressedAccountData>::new_init(
-            &crate::ID,
-            Some(address),
-            output_tree_index,
+            &crate::ID,           // Program ID that owns this account
+            Some(address),        // Derived address for the account
+            output_tree_index,    // Position in the state tree
```

Comment:
```suggestion
            output_tree_index,    // Index of the state tree in packed accounts
```
The actual position in the tree, the leaf index, is set in the light system program.